datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaTransportProvider.java

```diff
@@ -80,6 +81,9 @@ public KafkaTransportProvider(DatastreamTask datastreamTask, List<KafkaProducerW
       ErrorLogger.logAndThrowDatastreamRuntimeException(LOG, errorMessage, null);
     }
 
+    _preserveEventSourceTimestamp = Boolean.parseBoolean(datastreamTask.getDatastreams().get(0).getMetadata()
+            .get(DatastreamMetadataConstants.PRESERVE_EVENT_SOURCE_TIMESTAMP));
```

Comment:
can you change this to '.getOrDefault(DatastreamMetadataConstants.PRESERVE_EVENT_SOURCE_TIMESTAMP, Boolean.FALSE.toString())' so that the default value it should have is clear?

Reply:
Should it not be clear from Boolean.parseBoolean()'s behavior (returning false) if the config is not present? I can still change it if you want

Reply:
done